build_tools/release_automation.py

```diff
@@ -0,0 +1,262 @@
+    output_lines = []
+    with open(init_file, 'r') as f:
+        for line in f.readlines():
+            if line[:11] == '__version__':
+                output_lines.append('__version__ = \"' + version + '\"\n')
+            else:
+                output_lines.append(line)
+    with open(init_file, 'w') as f:
+        f.writelines(output_lines)
+
+def update_license(license_file, license_line, line_to_update):
+    """
+
+    :return:
+    """
+    with open(license_file, 'r') as f:
+        output_lines = f.readlines()
+        output_lines[line_to_update] = license_line
+    with open(license_file, 'w') as f:
+        f.writelines(output_lines)
+
+
+def update_acknowledgement_widget():
+    """
+
+    :return:
+    """
+    pass
+
+
+def prepare_release_notes(issues_list, repository, username, password):
+    """
+    Retrieving information from github and creating issue list for release notes
+    :return:
+    """
+    issue_titles = []
+    for issue in issues_list:
+        #WARNING: One can try running with auth but there is limitted number of requests
+        response = requests.get('https://api.github.com/repos/SasView/' + repository + '/issues/' + issue,
+                                auth=(username, password))
+        title = response.json()['title']
+        issue_title = f"#{issue}, {title}"
+        issue_titles.append(issue_title)
+    return issue_titles
+
+if __name__ == "__main__":
+    """
+    Setups init and license files
+    Generates zenodo doi and writes to the init file
+    Templates release notes
+    """
+
+    parser = argparse.ArgumentParser('Script to automate release process')
+    parser.add_argument('-v', '--sasview_version', required=True)
+    parser.add_argument('-s', '--sasmodels_version', required=True)
+    parser.add_argument('-z', '--zenodo', default=False)
+    parser.add_argument('-u', '--username', default=False)
+    parser.add_argument('-p', '--password', default=False)
+    parser.add_argument('-l', '--sasview_list', default=False, action=SplitArgs)
+    parser.add_argument('-m', '--sasmodels_list', default=False, action=SplitArgs)
+    args = parser.parse_args()
+
+    sasview_version = args.sasview_version
+    sasmodels_version = args.sasmodels_version
+    sasview_data['metadata']['title'] = 'SasView version '+ sasview_version
+    sasview_data['metadata']['description'] = sasview_version + ' release'
+    sasview_data['metadata']['related_identifiers'][0]['identifier'] = \
+        'https://github.com/SasView/sasview/releases/tag/v' + sasview_version
+
+    # Generates zenodo doi if zenodo api key is provided
+    new_doi = ''
+    if args.zenodo:
+        zenodo_api_key = args.zenodo
+        new_doi = generate_zenodo(sasview_data, zenodo_api_key)
+
+    update_sasview_init(sasview_version, new_doi)
+    update_sasmodels_init(sasmodels_version)
+
+    year = datetime.datetime.now().year
+    license_line = 'Copyright (c) 2009-' + str(year) + ', SasView Developers\n'
```

Comment:
Should probably end with `\n\n` otherwise it looks clumsy.

>Copyright (c) 2009-2020, SasView Developers
>All rights reserved.
>
>Redistribution and...
`


>